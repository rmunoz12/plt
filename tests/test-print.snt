@setup {

group A() {
    func void __init__() { }
};

group B(A) {
    func str __repr__() {
        return "Hello from class B!";
    }
};

group C(B) {

};

group A no_repr;
group B yes_repr;
group C inherited_repr;

}

@turns {

func void begin() {

    print(1); print("\n");
    print("abc"); print("\n");
    print(True); print(" "); print(False); print("\n");
    print(no_repr); print("\n");
    print(no_repr.__repr__()); print("\n");
    print(yes_repr); print("\n");
    print(inherited_repr); print("\n");

    end;
}

}
