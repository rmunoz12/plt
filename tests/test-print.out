1
abc
true false
<Group A instance>
<Group A instance>
Hello from class B!
Hello from class B!
<Group D instance>
